ui/src/app/utils/leo-runtime-initializer.tsx

```diff
@@ -202,13 +204,23 @@ export class LeoRuntimeInitializer {
   }
 
   private async createRuntime(): Promise<void> {
+    const { gcePersistentDisk } = diskStore.get();
+
     if (!this.targetRuntime) {
       // Automatic lazy creation is not supported; the caller must specify a target.
       let defaultRuntime = {
         ...runtimePresets.generalAnalysis.runtimeTemplate,
       };
       if (this.currentRuntime) {
-        defaultRuntime = applyPresetOverride(this.currentRuntime);
+        defaultRuntime =
+          this.currentRuntime.status === RuntimeStatus.Deleted
+            ? applyPresetOverride(
+                // The attached disk information is lost for deleted runtimes. In any case,
+                // by default we want to offer that the user reattach their existing disk,
+                // if any and if the configuration allows it.
+                maybeWithExistingDisk(this.currentRuntime, gcePersistentDisk)
```

Comment:
This is one part of the fix: combine a `gceConfig` with a `gcePersistentDisk` to create a `gceWithPdConfig`, if possible.